### DISCUSSION TITLE
Option to set the expiration date based on the Published date

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/publishpress-future#370] [Feature Requests]
> I would like to use this plugin to make a publication, with a specific category, expire AND have the expiration happen 30 days after the article is published.
> But I have a problem: I plan my post date, which can be in the future while the expiration date +30 days is calculated from the moment I place the post in the database.
> To be clear, I insert in the database the 30/11/2021 but the planned date is 8/12/2021. In this case, the plugin calculates the expiration date on 30/12/2021 and not on 8/1/2022.
> Is it possible to adjust this because it seems to me that the expiration date should run only after the publication is public.

**Comment by andergmartins** [comment]
We've implemented this for workflows.
